util/index.ts

```diff
@@ -0,0 +1,107 @@
+import inquirer, { Answers } from 'inquirer';
+import inquirerFuzzyPath, { FuzzyPathQuestionOptions } from 'inquirer-fuzzy-path';
+
+import { Lang } from '../resources/languages';
+
+import { run as findMissingTranslations } from './find_missing_translations';
+import { default as generateEffectIds } from './gen_effect_id';
+import { run as translateTimeline } from './translate_timeline';
+
+declare module 'inquirer' {
+  interface QuestionMap<T> {
+    fuzzypath: FuzzyPathQuestionOptions<T>;
+  }
+}
+
+const dataFilesMap = {
+  'effect_id.ts': generateEffectIds,
+} as const;
+
+inquirer.registerPrompt('fuzzypath', inquirerFuzzyPath);
+
+const run = () => {
+  inquirer.prompt([{
+    type: 'list',
+    name: 'action',
+    message: 'What do you want to do?',
+    choices: [
+      'Generate common data files',
+      'Translate Raidboss timeline',
+      'Find Raidboss missing translations',
+    ],
+  }]).then((answer: Answers) => {
+    if (answer.action === 'Generate common data files')
+      return generateDataFiles();
+    if (answer.action === 'Translate Raidboss timeline')
+      return translateTimelineFunc();
+    if (answer.action === 'Find Raidboss missing translations')
+      return findMissingTranslationsFunc();
+  }).catch(console.error);
+};
+
+const generateDataFiles = () => {
+  return inquirer.prompt([{
+    type: 'list',
+    name: 'choice',
+    message: 'Which data file do you want to generate?',
+    choices: Object.keys(dataFilesMap),
+  }]).then((answers: Answers) => {
+    if (answers.choice in dataFilesMap)
+      return dataFilesMap[answers.choice as keyof typeof dataFilesMap]?.();
+  });
+};
+
+const translateTimelineFunc = () => {
+  return inquirer.prompt([
+    {
+      type: 'fuzzypath',
+      name: 'timeline',
+      message: 'Input a valid timeline filename: ',
+      rootPath: 'ui/raidboss/data',
+    },
+    {
+      type: 'list',
+      name: 'locale',
+      message: 'Select a locale: ',
+      choices: [
+        'de',
+        'fr',
+        'ja',
+        'cn',
+        'ko',
+      ],
+    },
+  ]).then((answers: Answers) => {
+    if (answers.timeline && answers.locale)
+      return translateTimeline(answers as { timeline: string; locale: Lang });
```

Comment:
```suggestion
    const timeline = answers.timeline;
    const locale = answers.locale;
    if (timeline && locale)
      return translateTimeline({ timeline: timeline; locale: locale });
```

Alternatively, just call `return translateTimeline(answers);` and have the `translateTimeline` function handle missing params?